samza-log4j/src/main/java/org/apache/samza/logging/log4j/StreamAppender.java

```diff
@@ -279,8 +285,17 @@ protected Config getConfig() {
     return config;
   }
 
+  private void addDropProducerErrorsConfig() {
+    Map<String, String> newConfig = new HashMap<>();
+    if (config.get(DROP_PRODUCER_ERRORS) == null) {
```

Comment:
Does it makes sense to turn this on by default regardless of whatever value user has defined for `task.drop.producer.errors`?

Reply:
@PawasChhokra This was the intention. It shouldn't check if they've opted in for this, it should be enabled by default.

Reply:
Got it. That means I have to let go of the if statement and turn it on by default.